update stackpath provider documentation to include allocation resources and workload spec changes (#47)
stackpath/terraform-provider-stackpath: website/docs/r/compute_network_allocation_claim.html.markdown

---
layout: "stackpath"
page_title: "StackPath: stackpath_compute_network_allocation_claim"
sidebar_current: "docs-stackpath-resource-compute-network-allocation-claim"
description: |-
  IP allocation claim required to claim IP from an allocation.
---

# stackpath\_compute\_network\_allocation\_claim

> IP allocation claim used to claim IP from an allocation to network interfaces of StackPath computing workloads.

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

  # A DNS compatible label value that is unique to your stack. This value must
  # be RFC 1123 compliant (only contain "a-z", "0-9", "-", ".").
  slug = "allocation-claim-allocation-template"

  # allocation IP family, either IPv4 or IPv6
  ip_family = "IPv4"

  # allocation prefix length
  prefix_length = 32

  # allocation reclaim policy, only RETAIN action is supported from API
  reclaim_policy {
    action = "RETAIN"
  }

  allocation {
    # use selector to match to cityCode and allocation label to claim from.
    template {
      allocation_class = "stackpath-edge/unicast"
      ip_family = "IPv4"
      prefix_length = 32
      reclaim_policy {
        action = "RETAIN"
      }

      selectors {
        key = "cityCode"
        operator = "in"
        values = ["DFW"]
      }
    }
  }
}

```

## Argument Reference

* `name` - (Required) A human readable name.
* `slug` - (Required) A programmatic name for the network allocation claim.
* `labels` - (Optional) Key/value pairs of arbitrary label names and values that can be referenced as selectors.
* `annotations` - (Optional) Key/value pairs that define StackPath-specific network allocation configuration.

> * `ip_family` - (Required) A IP Family of the IPs being allocated. One of the IPv4 or IPv6 can be provided.

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

> * `prefix_length` - (Required) A Prefix length of IP allocation. Currently only 32 and 128

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

prefix length values are supported for IPv4 and IPv6 respectively.

> * `reclaim_policy` - (Required) A reclaim policy to be used for IP allocation. only RETAIN action is supported in reclaim policy specified in allocation resources being created from API.

Reviewer: ```suggestion
* `reclaim_policy` - (Required) A reclaim policy to be used for an IP allocation. Only the RETAIN action is supported in reclaim policy specified in allocation resources being created from API.
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

* `allocation` - (Required) An allocation for the claim can be defined in three mutually exclusive ways:

1. Directly via reference in resource name format.
2. Selecting across a set of existing allocations, allowing for the definition of "pools" of addresses to use for different purposes.
3. Via the definition of a `template`, which will create an allocation for the claim if one does not already exist. The allocation will be identified by a slug unique to the claim.

See [Allocation](#allocation) below for details.

### Allocation

`allocation` take the one of the following arguments:

> * `name` - (Optional) The name of allocation in resource name format.

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

> * `selector` - (Required) A selector to select existing allocation using allocation_class

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

> and match_expressions used to match allocation. See [Selector](#selector) below for details.

Reviewer: ```suggestion
and match_expressions used to match an allocation. See [Selector](#selector) below for details.
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

> * `template` - (Required) A allocation resource template, it creates an allocation using

Reviewer: ```suggestion
* `template` - (Required) An allocation resource template. It creates an allocation using the
```

I don't actually like that suggestion because the first "sentence" is just a fragment. It should really be "An allocation resource template creates an allocation...", but there is a pattern standing out that most of these start with descriptive sentence fragments. I'll trust you to use your judgment.

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466), hope new wording is fine.?

provided template if allocation does not exist. See [Template](#template) below for details.

### Selector

`selector` takes following arguments:

> * `allocation_class` - (Required) A IP allocation class to allocate IP from. Supported values are stackpath-edge/private and stackpath-edge/unicast.

Reviewer: ```suggestion
* `allocation_class` - (Required) An IP allocation class to allocate an IP address from. Supported values are stackpath-edge/private and stackpath-edge/unicast.
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

* `match_expressions` - (Required) A list of match expressions to match allocation. See [Match_Expressions](#match_expressions) below for details

### Match_Expressions

> `match_expressions` take the following arguments:

Reviewer: ```suggestion
`match_expressions` values take the following arguments:
```

Even though the value of match_expressions may be plural, it refers to a singular field. "[The filed named match_expressions] takes..." But that's incorrect too because of the data structure. This is how I arrived at "values take" for subject-verb agreement.

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

* `key` - (Required) The name of the data that a selector is based on.
* `operator` - (Required) A logical operator to apply to a selector. Only the "in" operator is supported.
* `values` - (Required) Data values to look for in a label selector.

### Template

`template` take the following arguments:

> * `allocation_class` - (Required) A IP allocation class to allocate IP from. Supported values are stackpath-edge/private and stackpath-edge/unicast.

Reviewer: ```suggestion
* `allocation_class` - (Required) An IP allocation class to allocate IP from. Supported values are stackpath-edge/private and stackpath-edge/unicast.
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

> * `ip_family` - (Required) A IP Family of the IPs being allocated. One of the IPv4 or IPv6 can be provided.

Reviewer: ```suggestion
* `ip_family` - (Required) An IP family of the IPs being allocated. One of IPv4 or IPv6 can be provided.
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

> * `prefix_length` - (Required) A Prefix length of IP allocation. Currently only 32 and 128

Reviewer: ```suggestion
* `prefix_length` - (Required) A prefix length of IP allocation. Currently only 32 and 128
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

prefix length values are supported for IPv4 and IPv6 respectively.

> * `reclaim_policy` - (Required) A reclaim policy to be used for IP allocation. only RETAIN action is supported in reclaim policy specified in allocation resources being created from API.

Reviewer: ```suggestion
* `reclaim_policy` - (Required) A reclaim policy to be used for IP allocation. Only RETAIN action is supported in reclaim policy specified in allocation resources being created from API.
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

> * `selectors` - (Required) A edge location selector that the network allocation applies to. See [Selectors](#selectors) below for details.

Reviewer: ```suggestion
* `selectors` - (Required) An edge location selector that the network allocation applies to. See [Selectors](#selectors) below for details.
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)

### Selectors

> `selectors` take the following arguments:

Reviewer: ```suggestion
`selectors` values take the following arguments:
```

Author: addressed in [eef298e](https://github.com/stackpath/terraform-provider-stackpath/pull/47/commits/eef298e528faf9da80cc5a21b0addb9357599466)
